pkg/controller/spinnakerservice/spinnakerservice_controller.go

```diff
@@ -34,8 +35,8 @@ func Add(mgr manager.Manager) error {
 }
 
 type deployer interface {
-	IsSpinnakerUpToDate(svc spinnakerv1alpha1.SpinnakerServiceInterface, config runtime.Object, hc *halconfig.SpinnakerConfig) (bool, error)
-	Deploy(svc spinnakerv1alpha1.SpinnakerServiceInterface, scheme *runtime.Scheme, config runtime.Object, hc *halconfig.SpinnakerConfig) error
+	IsSpinnakerUpToDate(ctx context.Context, svc spinnakerv1alpha1.SpinnakerServiceInterface, config runtime.Object, hc *halconfig.SpinnakerConfig) (bool, error)
+	Deploy(ctx context.Context, svc spinnakerv1alpha1.SpinnakerServiceInterface, scheme *runtime.Scheme, config runtime.Object, hc *halconfig.SpinnakerConfig) error
```

Comment:
Maybe we can have another struct act as a placeholder for common objects that need to be passed everywhere, like `context` and `SpinnakerServiceInterface`. In this way we could reduce the number of parameters on the functions.
Also `SpinnakerConfig` is created from `config runtime.Object`, maybe we can include the runtime object as part of the `SpinnakerConfig` struct?

Reply:
All parameters have a good reason to be here.  `context` is quite standard and expected. I actually like `SpinnakerConfig` and `config` separated as you can use the former without the latter. My comment on the signature was more for the transformers, change detectors and validators to make it easier for people to contribute. We may address that via good examples and documentation. I'll keep track of that in a separate issue.